Modules/vc-module-catalog/VirtoCommerce.CatalogModule.Data/ExportImport2/CatalogFullExportPagedDataSource.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.Search;
+using VirtoCommerce.CatalogModule.Core.Search;
+using VirtoCommerce.CatalogModule.Core.Services;
+using VirtoCommerce.ExportModule.Core.Model;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CatalogModule.Data.ExportImport
+{
+    public class CatalogFullExportPagedDataSource : IPagedDataSource
+    {
+        private class FetchResult
+        {
+            public int TotalCount;
+            public IEnumerable<IExportable> Result = Array.Empty<IExportable>();
+        }
+
+        private class ExportDataSourceState
+        {
+            public FetchResult FetchResult;
+            public SearchCriteriaBase SearchCriteria;
```

Comment:
Maybe we could make generic class with and use ATP to instantiate it on creation to exclude from constructor:
```
class ExportDataSourceState<TSearchCriteria> where TSearchCriteria: SearchCriteriaBase
{
public SearchCriteriaBase SearchCriteria = AbstractTypeFactory<TSearchCriteria>.TryCreateInstance();
}
```